src/main/webapp/scripts/analyze.js

```diff
@@ -0,0 +1,44 @@
+function analyzeVideo() {
+  const videoId = document.getElementById('video-url').value.split('v=')[1];
+  const baseUrl = window.location.origin;
+  let url = new URL('/sentiment', baseUrl);
+  url.searchParams.append('video-id', videoId);
+
+  fetch(url)
+  .then(response => {
+    if (response.status >= 200 && response.status <= 299) {
+      return response.json();
+    } else if (response.status >= 400 && response.status <= 499) {
+      throw InvalidVideoException(response.statusText);
+    } else {
+      throw ServletFailureException(response.statusText);
+    }
+  })
+  .then(videoAnalysis => {
+    if (videoAnalysis.dataAvailable) {
+      updateSentimentContainer(videoAnalysis.score);
+    } else {
+      updateSentimentContainer("We couldn't analyze this video! Try again.")
+    }
+  })
+  .catch(error => {
+    updateSentimentContainer(error.message);
```

Comment:
Could you also log into the console? `console.error(error)`

Reply:
if the servlet calls ```response.sendError()``` it logs it into the console already!

Reply:
Whoa. Didn't know that.